src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -1,38 +1,130 @@
 package org.constellation.domain.redownload
 
+import cats.effect.{ContextShift, LiftIO, Sync}
 import cats.effect.Concurrent
 import cats.effect.concurrent.Ref
 import cats.implicits._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import org.constellation.DAO
+import org.constellation.consensus.EdgeProcessor
+import org.constellation.domain.snapshotInfo.SnapshotInfoChunk
 import org.constellation.schema.Id
+import org.constellation.storage.RecentSnapshot
+import org.constellation.util.MajorityStateChooser.NodeSnapshots
+import org.constellation.util.{HealthChecker, MajorityStateChooser}
 
-class RedownloadService[F[_]]()(implicit F: Concurrent[F]) {
+import scala.concurrent.duration._
+
+class RedownloadService[F[_]](dao: DAO)(implicit F: Concurrent[F], C: ContextShift[F]) {
```

Comment:
Imo we could just have `cluster: Cluster[F]` instead of whole `dao: DAO`. We wanted to get rid of injecting DAO so better start doing it in the new code as services should get only these services which are needed instead of "everything" :P

Reply:
I was just removing and I realized that we need to call dao.readyPeers in recalculateMajoritySnapshot. Well need to remove peerInfo from dao into it's own service to refactor

Reply:
@buckysballs we have a `Cluster` for that

Reply:
Oh right , cool I'll use Cluster and HealthChecker